src/Core/Domain/Product/Image/ValueObject/NewConfigurableImage.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Image\Exception\ProductImageConstraintException;
+use SplFileObject;
+use function in_array;
+
+/**
+ * New not created image.
+ */
+final class NewConfigurableImage implements ConfigurableImageInterface
```

Comment:
Same here, I feel like this needs redesign.

Reply:
well I don't think that way - to change naming is always welcome. Maybe interface is not needed as well?  Probably yes,  but having such deeply separated value objects will be beneficial due to:

imagine we have `ProductImage` VO - it always brings filePath information and other information with it. But for example when we update image caption for same image we don't need anything related with file. In your case you suggest having same value object for both add and update cases? Will it actually be value object in such case?

Reply:
> But for example when we update image caption for same image we don't need anything related with file.

So maybe we should think differently? For me it seems as a totally different command, something like `UpdateImageCaptionCommand` or something like that.